src/librustdoc/html/static/js/main.js

```diff
@@ -54,6 +54,339 @@ function showMain() {
 window.rootPath = getVar("root-path");
 window.currentCrate = getVar("current-crate");
 
+// Tag deprecated entries in the current page and sidebar, and expose an updater
+function rustdocTagDeprecated() {
```

Comment:
All of these changes are unnecessary. If you have the document class set to hide deprecated items, then a rule in the CSS should hide all of them, there is no need to modify the existing HTML.